ddtrace/internal/processor/trace.py

```diff
@@ -58,13 +58,12 @@ class TraceSamplingProcessor(TraceProcessor):
 
     def process_trace(self, trace):
         # type: (List[Span]) -> Optional[List[Span]]
-        sampled_spans = [s for s in trace if s.sampled]
-        if len(sampled_spans) == 0:
-            log.info("dropping trace, %d spans unsampled", len(trace))
-            return None
+        for span in trace:
+            if span.sampled:
+                return trace
 
-        log.info("trace %d sampled (%d/%d spans sampled)", trace[0].trace_id, len(sampled_spans), len(trace))
```

Comment:
Hmm I think this information is still useful for debugging. I imagine a case where the user thinks that they have unsampled all the spans in a trace but really they missed one or two in which case this log line might be helpful.

This might be a little far-fetched and not worth it though - curious what others think! @brettlangdon, @majorgreys, @Yun-Kim 🙂

Reply:
should we update the debug log below in `SpanAggregator.on_span_finish` to say whether it was sampled or not?

Reply:
@brettlangdon if we start checking the sampling info in `SpanAggregator` then we might as well do away with `TraceSamplingProcessor`

Reply:
we can probably just look at the first span, all the spans should share the same `span.sampled` value no? and if they were different, I don't think that is even supported is it? the behavior would definitely be unexpected.

Reply:
>  all the spans should share the same span.sampled value no? 
Exactly, I think that makes the most sense here to just look at the first span instead of having to do all spans in the trace.

Reply:
It is supported unfortunately. Users can do `span.sampled = False` to unsample the whole trace 😕 

I don't like this functionality but this is what we have supported up to this point.

Reply:
is this how it worked previously?

Reply:
yeah it's been supported for as long as I can remember. Don't know if we've ever recommended it to users though.

Reply:
massive digression so feel free to ignore.

should the sampled yes/no be stored on the context? similar to origin or sampling priority.

as long as the current behavior is meant to be "if any span in this trace is not sampled, then the entire trace isn't sampled", then we can definitely optimize that a bit.

Reply:
> should the sampled yes/no be stored on the context? similar to origin or sampling priority.

I was thinking of a similar approach for even better performance.

Reply:
It should be pretty doable but has the potential to introduce performance problems in multithreaded apps because storing things in the context requires a lock.